.github/workflows/main.yml

```diff
@@ -7,107 +7,21 @@ on:
       - master
 
 jobs:
-  # Enforce the standard of g_malloc instead of malloc in the project
-  enforce-g_malloc:
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v2
-      with:
-        # Clone the whole branch, we have to fetch tags later
-        fetch-depth: 0
-
-    - name: enforce-g_malloc
-      run: "! git grep -P '(?<!g_)malloc' ':!.valgrind.suppressions' ':!.github/workflows/main.yml'"
-
   build:
-    strategy:
-      matrix:
-        CC:
-          - clang
-          - gcc
-        distro:
-          - alpine
-          - archlinux
-          - debian-stretch
-          - debian-buster
-          - fedora
-          - ubuntu-xenial
-          - ubuntu-bionic
-          - ubuntu-focal
-
-    env:
-      CC: ${{ matrix.CC }}
-      EXTRA_CFLAGS: "-Werror"
+    runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-      with:
-        # Clone the whole branch, we have to fetch tags later
-        fetch-depth: 0
-
-    # Fetch tags to determine proper version number inside git
-    - name: fetch tags
-      run: git fetch --tags
-      # We cannot pull tags with old distros, since there is no `.git`. See below.
-      if: "! (matrix.distro == 'ubuntu-bionic' || matrix.distro == 'ubuntu-xenial' || matrix.distro == 'debian-stretch')"
-
-    # The Github checkout Action doesn't support distros with git older than 2.18
-    # With git<2.18 it downloads the code via API and does not clone it via git :facepalm:
-    # To succeed the tests, we have to manually replace the VERSION macro
-    - name: fix version number for old distros
-      run: 'sed -i "s/-non-git/-ci-oldgit-$GITHUB_SHA/" Makefile'
-      if: "  (matrix.distro == 'ubuntu-bionic' || matrix.distro == 'ubuntu-xenial' || matrix.distro == 'debian-stretch')"
-
-    - name: build
-      run: make -j all dunstify test/test
-
-    - name: test
-      run: make -j test
+    - uses: actions/checkout@v4
 
-    - name: installation
-      run: ./test/test-install.sh
-
-    - name: valgrind memleaks
-      run: |
-        make clean
-        make -j test-valgrind
-
-    - name: coverage
+    - name: Install dependencies
       run: |
-        make clean
-        make -j test-coverage
-
-    - name: Generate coverage report
-      run: lcov -c -d . -o lcov.info
-      if: "matrix.CC == 'gcc'"
-
-    - name: Upload coverage to Codecov
-      uses: codecov/codecov-action@v1
-      with:
-        token: ${{ secrets.CODECOV_TOKEN }}
-        flags: unittests
-        name: ${{ matrix.distro }}-${{ matrix.CC }}
-        fail_ci_if_error: true
-      if: "matrix.CC == 'gcc'"
+        sudo apt install -y ninja-build python3-pip gcc pkg-config libwayland-dev libx11-dev libxss-dev libxinerama-dev libxrandr-dev libnotify-dev libdbus-1-dev libwayland-dev libpango1.0-dev
```

Comment:
Sort those arguments and split them into multi line. Otherwise this will be a start of a huge mess.

Also you need to add `export DEBIAN_FRONTEND=noninteractive` to it. See some other docker reference: https://github.com/moby/moby/issues/4032#issuecomment-34597177 (tianon is the docker Debian library image maintainer)

_(Additionally I don't think you need `sudo`, but I'm not sure about this. Keep in if you need to add sudo, the call should be `sudo env DEBIAN_FRONTEND=noninteractive apt-get` to pass the variable.)._

```suggestion
       export DEBIAN_FRONTEND=noninteractive
       apt-get install -y \
            gcc \
            libdbus-1-dev \
            libnotify-dev \
            libpango1.0-dev \
            libwayland-dev \
            libwayland-dev \
            libx11-dev \
            libxinerama-dev \
            libxrandr-dev \
            libxss-dev \
            ninja-build \
            pkg-config \
            python3-pip
```

Taken deliberately from [Dockerfile best practices](https://docs.docker.com/develop/develop-images/guidelines/#sort-multi-line-arguments). The "general" section is a very good guideline for basically any code project.